packages/loader/container-loader/api-report/container-loader.api.md

```diff
@@ -26,6 +29,45 @@ import { ITelemetryBaseLogger } from '@fluidframework/core-interfaces';
 import { ITelemetryLoggerExt } from '@fluidframework/telemetry-utils';
 import { IUrlResolver } from '@fluidframework/driver-definitions/internal';
 
+// @beta
+export interface ConnectionDiagnostics {
+    clientId?: string;
+    desiredConnectionMode?: ConnectionMode;
+    readonly stateProgression: [
+    /** Details about the connection while disconnected */
+        {
+        state: "disconnected";
+        readonly time: number;
+        readonly reason?: IConnectionStateChangeReason;
+        readonly autoReconnect: ReconnectMode;
+    }
+    /** Details about the connection while establishingConnection */
+    | {
+        state: "establishingConnection";
+        readonly time: number;
+        readonly reason?: IConnectionStateChangeReason;
```

Comment:
What is reason expected to convey here in contrast to the steps?

Reply:
Is the reason here basically the reason for disconnection?

Reply:
Every state transition actually has an internal reason.  It's the reason we moved into this state from the previous state.  So on "disconnected" state it's the disconnect reason, but for this state, some current examples (from the Loop doc I shared) include "Reconnecting due to: Switch to write" and "Reconnecting due to Error".  So you can see the disconnect reason is kind of embedded in there but it's rephrased.

We may revisit this, the reason field has evolved organically, it seems.